lib/elevator.js

```diff
@@ -72,10 +72,17 @@ exports.execute = function(command, options, callback) {
     debug('stderr: %s', stderr);
     debug('stdout: %s', stdout);
 
-    if (error && utils.doesErrorMeansElevationWasCancelled(error)) {
-      return callback.call(this, utils.ElevateCancelledError);
+    if (error) {
+      if (utils.doesErrorMeansElevationWasCancelled(error)) {
+        return callback.call(this, utils.ElevateCancelledError);
+      }
+
+      var elevationError = new Error('Couldn\'t elevate, exit code ' + error.code);
+      elevationError.description = error.message;
+      return callback(this, elevationError);
     }
 
+
     return callback.apply(this, arguments);
```

Comment:
Damn,

```js
return callback(this, elevationError);
```

should have been:

```js
return callback.call(this, elevationError);
```

I'll send a fix.

The `.call` and `.apply` function are able to invoke a function with a custom context. Their first argument is the context, which is `this` in this case, so we're sure the context of this function is preserved.

The second argument represents the arguments with which the function should be called. Both `.call` and `.apply` do the same thing, but they differ on the argument syntax they accept. `.call` works like this: `function.call(context, argument1, argument2, argument3, ...)`, while `.apply` accepts an array of arguments: `function.apply(this, [ argument1, argument2, argument3, ...])`